### PR TITLE
OS X Kodi.app script target replaced by a real app bundle target

### PR DESCRIPTION
**Pull request by HofiOne**
The goal was to create a bundled target that can be started and
debugged from Xcode just like the standalone Kodi binary.

Kept all of the 4 original build script steps, so the build process
itself does not changed at all, only the following changes made:
- XBMC_DEPENDS added to App-OSX.xcconfig common config file instead of
  defining per target
- otool -L calls replaced by otool -LX to eliminate the first summary
  line from the output result, that could cause the parsing of the result
  fail
- Kodi and Kodi.app compile switches unified based on Kodi settings
  mostly

### CODE REVIEW

File: Kodi.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -13335,36 +13328,45 @@
...
+		D6299AD61AFC2AA100E3059D /* Debug */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = E49910D1174E2E0000741B6D /* App-OSX.xcconfig */;
 			buildSettings = {
-				ARCHS = (
-					i386,
-					x86_64,
-				);
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CLANG_CXX_LANGUAGE_STANDARD = "c++0x";
```
Comment (Memphiz) [inline Kodi.xcodeproj/project.pbxproj]: we don't want to dupe/hardcode this its defined in App.xcconfig

File: Kodi.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -13335,36 +13328,45 @@
...
+				CLANG_CXX_LIBRARY = "libc++";
```
Comment (Memphiz) [inline Kodi.xcodeproj/project.pbxproj]: same

File: Kodi.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -13335,36 +13328,45 @@
...
+				CODE_SIGN_IDENTITY = "-";
```
Comment (Memphiz) [inline Kodi.xcodeproj/project.pbxproj]: where is this coming from?

File: Kodi.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -13335,36 +13328,45 @@
...
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_PREPROCESSOR_DEFINITIONS = (
+					"DEBUG=1",
+					"$(inherited)",
+				);
+				GCC_SYMBOLS_PRIVATE_EXTERN = NO;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				INFOPLIST_FILE = "";
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.7;
```
Comment (Memphiz) [inline Kodi.xcodeproj/project.pbxproj]: no thats not what we want - single source of truth here is App-OSX.xcconfig again

**Comment by Memphiz** [comment]
The xcode change is a mess. I know thats what xcode does - but you are using a pretty new xcode version. Are you sure that all the versions we mention in the readme are still working with those changes? Alot of definitions removed and others added in there.

What exactly is the benefit of this? Debugging the normal Kodi target is way faster because it doesn't run through all the otool step and copying steps. Thats why we have this in place. The app target is normally only there if we really wanna package it up into the deb.

**Comment by HofiOne** [comment]
These (unwanted?) changes made when I brought all target to the common default base (everywhere

**Comment by HofiOne** [comment]
Yes I see, it full of changes that xcode made when synced the target's settings. Instead of adding a new target i will change the type of the script target to app.

The real benefit is to be able to test the app instantly when a real bundle needed, just like when you have services, file associations, etc.  building the bunded app keeps just a few secs more time on my machine, i do not think it's a big overhead

**Comment by Memphiz** [comment]
Could you describe in baby speech again what the benefit is? I don't get it ...

**Comment by HofiOne** [comment]
What i'm planning to implement in the future is adding some services to Kodi, like "Add to Kodi Playlist" or "Play in Kodi", (because currently all file opening requests dropped by Kodi if once it is started)
Services like those are (re-)registered once the app bundle written to the disk and can be debugged more easier if  you start a real bundled app.

**Comment by Memphiz** [comment]
ok - then i suggest to make the needed changes to xcode as part of those projects (i guess you will do a pull request with those too). The most important thing is that we have most important stuff in the xcconfig files and mostly only the source files in xcode project. I know its a bitch to get right because xcode always changes stuff and no-one knows why. But we had issues with those things in the past where alterings to xcconfig weren't picked up correctly because something was defined in xcode by accident (and the xcconfig file was not set correctly in the project - we figured this out very very late in the process).

**Comment by HofiOne** [comment]
wouldn't it be better to do this step by step? right because of the problems you mentioned, a separate change like this can be reviewed much easier.

could you check this please

https://github.com/HofiOne/xbmc/commit/1f4ea527c62e87a6eed25406fc4a9702acecc008

that would be the version that just changes the type of the Kodi.app target to native app instead of dropping the old script only version and creating a new native one.

**Comment by HofiOne** [comment]
I did the same thing firsttime :) if you keep the cursor for a sec above the list elements you'll get a hint with the full path which  one is under the selection 

i'm searching for a solution right now how to force the correct one by default (as modifying manually seems storing as per/user state setting)

**Comment by HofiOne** [comment]
shared scheme might give a solution, but trying to look for something better  first...

**Pull request closed without merging**